api/envoy/api/v2/route/route.proto

```diff
@@ -437,6 +439,22 @@ message RouteAction {
     google.protobuf.Duration per_try_timeout = 3 [(gogoproto.stdduration) = true];
   }
 
+  // Specifies the idle timeout for the route. If not specified, there is no
+  // stream level timeout. This is distinct to :ref:`timeout
+  // <envoy_api_field_route.RouteAction.timeout>`, which provides an upper bound
+  // on the upstream response time; :ref:`idle_timeout
+  // <envoy_api_field_route.RouteAction.idle_timeout>` instead bounds the amount
+  // of time the request's stream may be idle.
+  //
+  // After header decoding, and assuming that this is not a WebSocket upgrade,
+  // the idle timeout will apply on downstream and upstream request events. Each
+  // time an encode/decode event for headers or data is processed for the
+  // stream, the timer will be reset. If the timeout fires, the stream is
+  // terminated with a 408 Request Timeout error code if no upstream response
+  // header has been received, otherwise a stream reset occurs.
```

Comment:
At some point when we have priorities properly implemented we may need to take into account stream starvation here (if HighPriStream3 is consuming all outgoing bandwidth but LowPriStream5 is otherwise ready to write it shouldn't time out).  Thankfully not something we have to worry about now.

Brainstorming other failure modes, if we have connections with very many streams, we might hit problems where the progress is super bunchy and causes artificial timeouts.  I think basically when there's space in the write buffer, every stream is informed and can do a full read and write, and then won't be called upon again until all those writes drain.    I don't know if there are enough users with backed up connections and many streams that it's worth calling out, but it's food for thought.

Reply:
I think it's probably going to confuse more than help to call this out, even though I agree, at extreme queueing scenarios this can happen.